pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/AbstractJavaAccessNode.java

```diff
@@ -135,4 +142,46 @@ private void setModifier(boolean enable, int mask) {
     public boolean isPackagePrivate() {
         return !isPrivate() && !isPublic() && !isProtected();
     }
+
+    @Override
+    public List<ASTAnnotation> getDeclaredAnnotations() {
```

Comment:
I just remembered, that ASTMethodDeclaration already implements a method with the same signature. We could reuse that implementation, which is simpler than this one, and delete it there

Reply:
`getDeclaraedAnnotations` in ASTMethodDeclaration just finds node's siblings（generally it's OK）. But we should find `ASTAnnotation`nodes before the node actually. I'm not sure which method we should use.

Reply:
IMO just finding `ASTAnnotation `type siblings of the node is OK according to the grammar.

Reply:
The grammar doesn't allow annotations to be siblings of any annotateable node and go after it, so the order check is currently pointless (may be needed in the future if the Java grammar ever allows annotations afterwards for a separate semantic). In the meantime, I'm pro to stay with the simplest implementation and avoid duplication.